source/directives/uiTreeNode.js

```diff
@@ -337,23 +398,27 @@
                   dragElm[0].style.display = displayElm;
                 }
 
+                //This checks if angularUiTree attributes are found on element.
                 outOfBounds = !UiTreeHelper.elementIsTreeNodeHandle(targetElm) &&
-                              !UiTreeHelper.elementIsTreeNode(targetElm) &&
-                              !UiTreeHelper.elementIsTreeNodes(targetElm) &&
-                              !UiTreeHelper.elementIsTree(targetElm) &&
-                              !UiTreeHelper.elementIsPlaceholder(targetElm);
+                    !UiTreeHelper.elementIsTreeNode(targetElm) &&
+                    !UiTreeHelper.elementIsTreeNodes(targetElm) &&
+                    !UiTreeHelper.elementIsTree(targetElm) &&
+                    !UiTreeHelper.elementIsPlaceholder(targetElm);
 
-                // Detect out of bounds condition, update drop target display, and prevent drop
+                //Detect out of bounds condition, update drop target display, and prevent drop, also reset parent to source.
                 if (outOfBounds) {
 
-                  // Remove the placeholder
+                  //Remove the placeholder.
                   placeElm.remove();
 
-                  // If the target was an empty tree, replace the empty element placeholder
+                  //If the target was an empty tree, replace the empty element placeholder.
                   if (treeScope) {
                     treeScope.resetEmptyElement();
                     treeScope = null;
                   }
+
+                  //Reset parent to source parent.
```

Comment:
This was added to address the following problem:

I have two trees, one is clone enabled and used to drag items into the other tree.  The clone enabled tree is a noDrop tree.  So if I drag a node out of the cloneEnabled, hover it over the other tree, and then drag it back to the cloneEnabled (also a noDrop) tree and drop it, it is added to the other tree.  The reason this happens is when the node hovers over a tree that accepts drops the parent tree is set, so when I leave that tree and go over to a noDrop tree and drop the node, it allows the drop logic to fire because outOfBounds is false (because I am in a ui-tree) but the parent is still set to the last tree I hovered over that allowed drops.

I hope that makes sense :/